electrum: broken on master because of dnspython update

dnspython was updated to 2.0.0 in master in a7b5b18ab92f4f4b303d3d3526b91a28f627c68c and this broke electrum.

Unfortunately loosening the version check does not fix the issue as electrum fails with the following error while running tests:

```
=================================== FAILURES ===================================
_________________ TestDnsSec.test_python_validate_rrsig_ecdsa __________________

self = <electrum.tests.test_dnssec.TestDnsSec testMethod=test_python_validate_rrsig_ecdsa>

    def test_python_validate_rrsig_ecdsa(self):
        rrset = dns.rrset.from_text("getmonero.org.", 3599, 1, 48,
                                    "257 3 13 mdsswUyr3DPW132mOi8V9xESWE8jTo0d xCjjnopKl+GqJxpVXckHAeF+KkxLbxIL fDLUT0rAK9iUzy1L53eKGQ==",
                                    "256 3 13 koPbw9wmYZ7ggcjnQ6ayHyhHaDNMYELK TqT+qRGrZpWSccr/lBcrm10Z1PuQHB3A zhii+sb0PYFkH1ruxLhe5g==")
        rrsig = dns.rdtypes.ANY.RRSIG.RRSIG.from_text(1, 46,
            dns.tokenizer.Tokenizer("DNSKEY 13 2 3600 20180612115508 20180413115508 2371 getmonero.org. SSjtP2jCtXPukps7E3kum709xq2TH6Lt Ur32UhE7WKwSUfLTZ4EAoD5g22mi1fpB GDGb30kCMndDVjnHAEBDWw=="))
        keys = {dns.name.Name([b'getmonero', b'org', b'']): rrset}
        origin = None
        now = 1527185178.7842247

        # 'None' means it is valid
>       self.assertEqual(None, dnssec.python_validate_rrsig(rrset, rrsig, keys, origin, now))

electrum/tests/test_dnssec.py:21:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

rrset = <DNS getmonero.org. IN DNSKEY RRset: [<257 3 13 mdsswUyr3DPW132mOi8V9xESWE8jTo0d xCjjnopKl+GqJxpVXckHAeF+KkxLbxIL fDLU...Uzy1L53eKGQ==>, <256 3 13 koPbw9wmYZ7ggcjnQ6ayHyhHaDNMYELK TqT+qRGrZpWSccr/lBcrm10Z1PuQHB3A zhii+sb0PYFkH1ruxLhe5g==>]>
rrsig = <DNS IN RRSIG(DNSKEY) rdata: DNSKEY 13 2 3600 20180612115508 20180413115508 2371 getmonero.org. SSjtP2jCtXPukps7E3kum709xq2TH6Lt Ur32UhE7WKwSUfLTZ4EAoD5g22mi1fpB GDGb30kCMndDVjnHAEBDWw==>
keys = {<DNS name getmonero.org.>: <DNS getmonero.org. IN DNSKEY RRset: [<257 3 13 mdsswUyr3DPW132mOi8V9xESWE8jTo0d xCjjnopKl...zy1L53eKGQ==>, <256 3 13 koPbw9wmYZ7ggcjnQ6ayHyhHaDNMYELK TqT+qRGrZpWSccr/lBcrm10Z1PuQHB3A zhii+sb0PYFkH1ruxLhe5g==>]>}
origin = None, now = 1527185178.7842247

    def python_validate_rrsig(rrset, rrsig, keys, origin=None, now=None):
        from dns.dnssec import ValidationFailure, ECDSAP256SHA256, ECDSAP384SHA384
>       from dns.dnssec import _find_candidate_keys, _make_hash, _is_ecdsa, _is_rsa, _to_rdata, _make_algorithm_id
E       ImportError: cannot import name '_to_rdata' from 'dns.dnssec' (/nix/store/nh4fj4kqi894476z606lj6a2003vm3i9-python3.8-dnspython-2.0.0/lib/python3.8/site-packages/dns/dnssec.py)

electrum/dnssec.py:69: ImportError
__________________ TestDnsSec.test_python_validate_rrsig_rsa ___________________
```

Upstream issue: https://github.com/spesmilo/electrum/issues/6538

I suggest downgrading dnspython back to 1.16 in nixpkgs or wait for an upstream to fix the issue (might not happen before 20.09).

--- a/pkgs/applications/misc/electrum/default.nix
+++ b/pkgs/applications/misc/electrum/default.nix
@@ -3,7 +3,6 @@
 , fetchFromGitHub
 , wrapQtAppsHook
 , python3
-, python3Packages
 , zbar
 , secp256k1
 , enableQt ? true
@@ -21,6 +20,20 @@
 
 let
   version = "4.0.2";
+
+  # electrum is not compatible with dnspython 2.0.0 yet
+  # use the latest 1.x release instead
+  py = python3.override {
+    packageOverrides = self: super: {
+      dnspython = super.dnspython.overridePythonAttrs (oldAttrs: rec {
+        version = "1.16.0";
+        src = oldAttrs.src.override {
+          inherit version;
+          sha256 = "36c5e8e38d4369a08b6780b7f27d790a292b2b08eea01607865bf0936c558e01";
+        };
+      });
+    };
+  };
 
   libsecp256k1_name =
     if stdenv.isLinux then "libsecp256k1.so.0"
@@ -45,7 +58,7 @@ let
   };
 in
 
-python3Packages.buildPythonApplication {
+py.pkgs.buildPythonApplication {
   pname = "electrum";
   inherit version;
 
@@ -61,7 +74,7 @@ python3Packages.buildPythonApplication {
 
   nativeBuildInputs = stdenv.lib.optionals enableQt [ wrapQtAppsHook ];
 
-  propagatedBuildInputs = with python3Packages; [
+  propagatedBuildInputs = with py.pkgs; [
     aiohttp
     aiohttp-socks
     aiorpcx
@@ -116,7 +129,7 @@ python3Packages.buildPythonApplication {
     wrapQtApp $out/bin/electrum
   '';
 
-  checkInputs = with python3Packages; [ pytest ];
+  checkInputs = with py.pkgs; [ pytest ];
 
   checkPhase = ''
     py.test electrum/tests

